src/utils/embeddableUrl.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  FACEBOOK_EMBED_URL,
+  TWITCH_EMBED_URL,
+  TWITCH_SHORT_URL,
+  VIMEO_EMBED_URL,
+  VIMEO_SHORT_EVENT_URL,
+  YOUTUBE_EMBED_URL,
+  YOUTUBE_SHORT_URL_STRING,
+} from "settings";
+
+const setParamsIteratively = (urlObject: URL, urlParams?: string[]) => {
+  urlParams?.forEach((param) => {
+    const [key, value] = param.split("=");
+    urlObject.searchParams.set(key, value);
+  });
+};
```

Comment:
I think object of the type https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams with utility methods can serve better than manually dealing with `string[]`

Reply:
done in 741d64c